### PR TITLE
feat: ketch consent manager plugin

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-plugins/__tests__/oneTrustConsentManager/index.test.ts [hunk 1]
```diff
@@ -115,16 +118,16 @@ describe('Plugin - OneTrust', () => {
...
       state,
       destConfig,
       defaultLogger,
     );
     expect(isDestinationConsented).toBeTruthy();
   });
   it('should return false if destination categories are not consented', () => {
-    state.consents.consentProviderInitialized.value = true;
-    state.consents.allowedConsents.value = {
+    state.consents.data.value.initialized = true;
```
Comment (bardisg) [inline packages/analytics-js-plugins/__tests__/oneTrustConsentManager/index.test.ts:129]: do not mutate the data object properties. you need to createnew data object and update it so refference is new and  any effect run. elsewise effects tracking the data object will not run

File: packages/analytics-js/src/components/core/Analytics.ts [hunk 2]
```diff
@@ -298,12 +298,11 @@ class Analytics implements IAnalytics {
     this.userSessionManager?.init(this.clientDataStore);
 
     // Initialize consent manager
-    if (state.consents.activeConsentProviderPluginName.value) {
+    if (state.consents.activeConsentManagerPluginName.value) {
       this.pluginsManager?.invokeSingle(
         `consentManager.init`,
         state,
-        this.pluginsManager,
-        batch,
```
[the remark this answers is not in the document]
Comment (bardisg) [reply packages/analytics-js/src/components/core/Analytics.ts:306]: read through the rest of the code and batch is not used anymore, please ignore

File: packages/analytics-js-plugins/src/ketchConsentManager/index.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+const KetchConsentManager = (): ExtensionPlugin => ({
+  name: pluginName,
+  deps: [],
+  initialize: (state: ApplicationState) => {
+    state.plugins.loadedPlugins.value = [...state.plugins.loadedPlugins.value, pluginName];
+  },
+  consentManager: {
+    init(state: ApplicationState, storeManager?: IStoreManager, logger?: ILogger): void {
+      const updateConsentStateFromData = (ketchConsentData: KetchConsentData) => {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/src/ketchConsentManager/index.ts]: Addressed.

File: packages/analytics-js-plugins/src/oneTrustConsentManager/index.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-param-reassign */
+import {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/src/oneTrustConsentManager/index.ts:2]: This was just renamed and also a bit refactored per the new flow.

File: packages/analytics-js/src/components/pluginsManager/PluginsManager.ts [hunk 5]
```diff
@@ -144,29 +144,14 @@ class PluginsManager implements IPluginsManager {
...
-      );
-    }
+    const supportedConsentManagerPlugins: string[] = Object.values(ConsentManagersToPluginNameMap);
+    pluginsToLoadFromConfig = pluginsToLoadFromConfig.filter(
+      pluginName =>
+        !(
+          pluginName !== state.consents.activeConsentManagerPluginName.value &&
+          supportedConsentManagerPlugins.includes(pluginName)
+        ),
+    );
```
Comment (MoumitaM) [inline packages/analytics-js/src/components/pluginsManager/PluginsManager.ts:154]: Why do we want to compute this if none of the consent managers is enabled by the customer through the load option?

**Review by MoumitaM — CHANGES_REQUESTED** [review]

File: packages/analytics-js/src/components/pluginsManager/PluginsManager.ts (hunk 5 above)
Comment (saikumarrs) [reply packages/analytics-js/src/components/pluginsManager/PluginsManager.ts:154]: By default, all the consent manager plugins are added. If no consent manager is configured, all of them must be filtered.

File: packages/analytics-js-plugins/src/oneTrustConsentManager/index.ts [hunk 6]
```diff
@@ -33,61 +41,45 @@ const OneTrust = (): ExtensionPlugin => ({
       const deniedConsentIds: string[] = [];
 
       // Get the groups(cookie categorization), user has created in one trust account.
-      const oneTrustAllGroupsInfo: OneTrustGroup[] = (globalThis as any).OneTrust.GetDomainData()
-        .Groups;
+      const oneTrustAllGroupsInfo: OneTrustGroup[] = (
+        globalThis as any
+      ).OneTrustConsentManager.GetDomainData().Groups;
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/src/oneTrustConsentManager/index.ts]: Addressed.

File: packages/analytics-js-plugins/src/oneTrustConsentManager/index.ts [hunk 7]
```diff
@@ -1,28 +1,36 @@
...
-    getConsentInfo(logger?: ILogger): ConsentInfo {
-      // In case OneTrust SDK is not loaded before RudderStack's JS SDK
-      // it will be treated as Consent manager is not initialized
-      if (!(globalThis as any).OneTrust || !(globalThis as any).OnetrustActiveGroups) {
-        logger?.error(ONETRUST_ACCESS_ERROR(ONETRUST_PLUGIN));
-        return { consentProviderInitialized: false };
+  consentManager: {
+    init(state: ApplicationState, storeManager?: IStoreManager, logger?: ILogger): void {
+      if (
+        !(globalThis as any).OneTrustConsentManager ||
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/src/oneTrustConsentManager/index.ts]: Addressed.

File: packages/analytics-js/src/components/pluginsManager/PluginsManager.ts (hunk 5 above)
Comment (MoumitaM) [reply packages/analytics-js/src/components/pluginsManager/PluginsManager.ts:154]: Got it.

**Review by MoumitaM — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Pull request merged**